cedar-policy/src/api.rs

```diff
@@ -1709,6 +1726,14 @@ impl PolicySet {
         Ok(())
     }
 
+    /// Remove a `Template` from the `PolicySet`
+    pub fn remove_template(&mut self, template_id: PolicyId) -> Result<(), PolicySetError> {
+        match self.templates.remove(&template_id) {
+            Some(_) => Ok(()),
+            None => Err(PolicySetError::RemoveTemplateError(template_id)),
+        }
```

Comment:
same here, I think we need to also remove from `self.ast` and add testing that authz queries are affected